test/EH/optional-catch-binding.js

```diff
@@ -0,0 +1,65 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+// Tests optional catch binding syntax
+
+WScript.LoadScriptFile("..\\UnitTestFramework\\UnitTestFramework.js");
+
+var tests = [
+  {
+    name: "Try-catch with no catch binding",
+    body: function() {
+      try {} catch {}
+    },
+  },
+  {
+    name: "Try-catch-finally with no catch binding",
+    body: function() {
+      try {} catch {} finally {}
+    },
+  },
+  {
+    name: "Try-catch with no catching binding lexical scope",
+    body: function() {
+      let x = 1;
+      let ranCatch = false;
+
+      try {
+        x = 2;
+        throw new Error();
+      } catch {
+        let x = 3;
+        let y = true;
+        ranCatch = true;
+      }
+
+      assert.isTrue(ranCatch, 'executed `catch` block');
+      assert.areEqual(x, 2);
+
+      assert.throws(function() { y; }, ReferenceError);
+    },
+  },
+  {
+    name: "Optional catch must not have empty parens",
+    body: function() {
+      assert.throws(function() { eval("try {} catch () {}"); }, SyntaxError);
+    },
+  },
+  {
+    name: "Error are correctly thrown from catch",
+    body: function() {
+      class Err {}
+      assert.throws(function() {
+        try {
+          throw new Error();
+        } catch {
+          throw new Err();
```

Comment:
is there a rethrow syntax as well?

Reply:
Taking a look at the proposal it doesn't look like it.  Seems like having `throw;` would have been interesting (like C# does), but it seems like if you want to rethrow then you'll need the parameter binding.

Reply:
Personally I like that you still have to be explicit if you want to rethrow, I always found `throw;` in C++ to be kind of obtuse.